src/contexts/Balance/Provider.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Provider } from '@ethersproject/abstract-provider';
+
+// Constants
+import { WPOKT_ADDRESS } from 'constants/index';
+
+// Context
+import { Web3Context } from 'contexts/Web3';
+import { BalanceContext } from './Context';
+
+// Utils
+import { getERC20Contract } from 'utils';
+
+export const BalanceProvider: React.FC = ({ children }) => {
+  const { address, provider } = React.useContext(Web3Context);
+  const [wpoktBalance, setWpoktBalance] = React.useState<string>();
+
+  const getBalance = async (provider: Provider, tokenAddress: string, userAddress: string): Promise<string> => {
+    const tokenContract = getERC20Contract(provider, tokenAddress);
+    try {
+      const balance = await tokenContract.balanceOf(userAddress);
+      return balance.toString();
+    } catch (e) {
+      console.error(e);
+      return '0';
+    }
+  };
+
+  const fetchBalances = React.useCallback(
+    async (userAddress: string, provider: Provider) => {
+      const balance = await getBalance(provider, WPOKT_ADDRESS as string, userAddress);
+      setWpoktBalance(balance);
+    },
+    // eslint-disable-next-line
+    [wpoktBalance],
+  );
+
+  React.useEffect((): (() => void) => {
+    if (address && provider) {
+      fetchBalances(address, provider);
+      const refreshInterval = setInterval(() => fetchBalances(address, provider), 10000);
+      return () => clearInterval(refreshInterval);
+    }
+    return () => null;
+    // eslint-disable-next-line
```

Comment:
Uhm haha should this be here?

Reply:
Whoops, yeah, shouldn't be there. Fixed in fcdfa813a94ff817747368a3058c107b850ab8c5